core/textile/mailbox.go

```diff
@@ -0,0 +1,45 @@
+package textile
+
+import (
+	"context"
+	"errors"
+
+	"github.com/FleekHQ/space-daemon/core/keychain"
+	crypto "github.com/libp2p/go-libp2p-crypto"
+	"github.com/textileio/go-threads/core/thread"
+	"github.com/textileio/textile/api/users/client"
+)
+
+type UsersClient interface {
+	SendMessage(ctx context.Context, from thread.Identity, to thread.PubKey, body []byte) (msg client.Message, err error)
+	SetupMailbox(ctx context.Context) (mailbox thread.ID, err error)
+}
+
+func (tc *textileClient) SendMessage(ctx context.Context, recipient string, body []byte) (*client.Message, error) {
+	kc := keychain.New(tc.store)
+	var privateKey crypto.PrivKey
+	var err error
+	if privateKey, _, err = kc.GetStoredKeyPairInLibP2PFormat(); err != nil {
+		return nil, err
+	}
+	id := thread.NewLibp2pIdentity(privateKey)
+
+	b := []byte(recipient)
```

Comment:
This is prone to encoding issues. Maybe instead of doing the conversion here, `SendMessage` could receive a `crypto.PubKey` instead of a string.

Reply:
Fixed in [e5c0fc5](https://github.com/FleekHQ/space-daemon/pull/108/commits/e5c0fc5a4cf56c80be4552aa84143ad8b4709fc0).